include/OpenSpaceToolkit/Astrodynamics/Flight/System/SatelliteSystemBuilder.hpp

```diff
@@ -0,0 +1,138 @@
+
+    SatelliteSystemBuilder& setDryMass(const Mass& aDryMass);
+
+    /// @brief                  Set the geometry
+    ///
+    /// @param                  [in] aGeometry A geometry
+    /// @return                 The builder
+
+    SatelliteSystemBuilder& setGeometry(const Composite& aGeometry);
+
+    /// @brief                  Set the inertia tensor
+    ///
+    /// @param                  [in] aInertiaTensor A inertia tensor
+    /// @return                 The builder
+
+    SatelliteSystemBuilder& setInertiaTensor(const Matrix3d& anInertiaTensor);
+
+    /// @brief                  Set the cross-sectional surface area
+    ///
+    /// @param                  [in] aCrossSectionalSurfaceArea A cross-sectional surface area
+    /// @return                 The builder
+
+    SatelliteSystemBuilder& setCrossSectionalSurfaceArea(const Real& aCrossSectionalSurfaceArea);
+
+    /// @brief                  Set the drag coefficient
+    ///
+    /// @param                  [in] aDragCoefficient A drag coefficient
+    /// @return                 The builder
+
+    SatelliteSystemBuilder& setDragCoefficient(const Real& aDragCoefficient);
+
+    /// @brief                  Set the propulsion system
+    ///
+    /// @param                  [in] aPropulsionSystem A Propulsion System
+    /// @return                 The builder
+
+    SatelliteSystemBuilder& setPropulsionSystem(const PropulsionSystem& aPropulsionSystem);
+
+    /// @brief                  Builds the satellite system
+    ///
+    /// @return                 A new satellite system
+    const SatelliteSystem build() const;
+
+    /// @brief                  Destructor
+
+    ~SatelliteSystemBuilder();
+
+    /// @brief                  Clone satellite system builder
+    ///
+    /// @return                 Pointer to cloned satellite system builder
+
+    SatelliteSystemBuilder* clone() const;
+
+    /// @brief                  Output stream operator
+    ///
+    /// @param                  [in] anOutputStream An output stream
+    /// @param                  [in] aSatelliteSystemBuilder A satellite system builder
+    /// @return                 A reference to output stream
+
+    friend std::ostream& operator<<(
+        std::ostream& anOutputStream, const SatelliteSystemBuilder& aSatelliteSystemBuilder
+    );
+
+    /// @brief                  Print satellite system builder
+    ///
+    /// @param                  [in] anOutputStream An output stream
+    /// @param                  [in] (optional) displayDecorators If true, display decorators
+
+    void print(std::ostream& anOutputStream, bool displayDecorator = true) const;
+
+   private:
+    Mass dryMass_ = Mass::Undefined();
+    Composite geometry_ = Composite::Undefined();
+    Matrix3d inertiaTensor_;
+    Real crossSectionalSurfaceArea_ = Real::Undefined();
+    Real dragCoefficient_ = Real::Undefined();
+    PropulsionSystem propulsionSystem_ = PropulsionSystem::Undefined();
```

Comment:
This is what I wrote down from the meeting in which we went over the items:

> Have everything Undefined by default
> Fail if something is missing

This way we have:
- An easy way to create system (only with the info we required)
- Something that will not use info with you not being aware

What do you think of the following: 
- we keep it as is (i.e. undefined by default)
- I add a `withDefaultValues()` so that one can populate everything without caring, then modify accordingly.

I think that's what you were suggesting right?